service/interpreter/temporal/workflowProvider.go

```diff
@@ -291,3 +298,39 @@ func (w *workflowProvider) GetUnhandledSignalNames(ctx interpreter.UnifiedContex
 	}
 	return workflow.GetUnhandledSignalNames(wfCtx)
 }
+
+func (w *workflowProvider) NewSelector(ctx interpreter.UnifiedContext) interpreter.UnifiedSelector {
+	wfCtx, ok := ctx.GetContext().(workflow.Context)
+	if !ok {
+		panic("cannot convert to temporal workflow context")
+	}
+	return &selector{
+		selector: workflow.NewSelector(wfCtx),
+	}
+}
+
+func (s *selector) Select(ctx interpreter.UnifiedContext) {
+	wfCtx, ok := ctx.GetContext().(workflow.Context)
+	if !ok {
+		panic("cannot convert to temporal workflow context")
+	}
+	s.selector.Select(wfCtx)
+}
+
+func (s *selector) ReceiveSignalValueBlocking(ctx interpreter.UnifiedContext, signalName string) interface{} {
+	wfCtx, ok := ctx.GetContext().(workflow.Context)
+	if !ok {
+		panic("cannot convert to temporal workflow context")
+	}
+
+	var signalValue interface{}
```

Comment:
I am not entirely sure this will work if you pass an interface{} into the receive API. Maybe it will decoded into a string or something. But the better way (Go way) is to ask caller to provide a pointer like in `ReceiveAsync`